sources/osgShader/Compiler.js

```diff
@@ -627,21 +629,14 @@ define( [
 
             var samplerName = 'Texture' + unit.toString();
             var textureSampler = this.getVariable( samplerName );
-
             if ( textureSampler === undefined ) {
-
-                if ( texture.className() === 'Texture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                } else if ( texture.className() === 'TextureCubeMap' ) {
-                    textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
-                } else if ( texture.className() === 'ShadowTexture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                    // return now to prevent creation of useless FragTexCoord
-                    //( shadow creates its own texcoord)
-                    return;
-                }
-
-
+                textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
+            } else if ( texture.className() === 'TextureCubeMap' ) {
+                textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
+            } else if ( texture.className() === 'ShadowTexture' ) {
```

Comment:
- *!textureSampler* :+1:  
- *broken* :seems a if disapeared here, will fix 
- *why do we always test for getVariable when we only do a getOrCreate just after?* Cannot Agree More ! Especially if we already does that in other production code... Those "Declare" are not flexible at all.